GliaWidgets/SwiftUI/Components/SnackBarView/SerialQueue.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+
+/// Represents queue with serial strategy. Not thread-safe.
+final class SerialQueue {
+    /// Adds operation to end of  serial queue
+    func addOpeation(_ op: Operation) {
+        let needToRun = queue.isEmpty
+        queue.append(op)
+        if needToRun {
+            run()
+        }
+    }
+
+    // MARK: - Private
+
+    private var queue = [Operation]()
+
+    private func run() {
+        guard queue.isEmpty == false else { return }
+
+        let op = queue[0]
+        op.didFinish = { [weak self] in
+            self?.queue.removeFirst()
+            self?.run()
+        }
+        op.run()
+    }
+}
+
+extension SerialQueue {
+    final class Operation {
+        var didFinish: (() -> Void)?
+        let action: (_ done: @escaping() -> Void) -> Void
+
+        init(action: @escaping (_ done: @escaping () -> Void) -> Void) {
+            self.action = action
+        }
+
+        func run() {
+            action(didFinish ?? {})
+        }
+    }
+}
+
+extension SerialQueue.Operation {
+    static func timeout(seconds: Int) -> Self {
+        .init(
+            action: { done in
+                DispatchQueue.main.asyncAfter(
```

Comment:
This delay should be achieved via reactive scheduler or injected closure. Otherwise it will hurt unit tests.

Reply:
it was and example, move into docs - a1759c3d75ec27d5d599b01c8f44716d567cf54f